src/app/login/login.component.ts

```diff
@@ -13,23 +13,46 @@ export class LoginComponent implements OnInit {
   providers = AuthProvider;
 
   constructor(
-    private route: ActivatedRoute,
     private router: Router,
-    private userInfoService: UserInfoService
+    private userInfoService: UserInfoService,
+    private problemListService: ProblemListService
   ) {}
 
   ngOnInit(): void {}
 
-  async login($event) {
-    const idToken = await $event.getIdToken();
-    const userInfo: IUserInfo = {
-      displayName: $event.displayName,
-      email: $event.email,
-      photoUrl: $event.photoURL,
-      uid: $event.uid,
-      idToken
-    };
-    this.userInfoService.setUserInfo(userInfo);
-    this.router.navigate(['/problem']);
+  login() {
+    this.userInfoService.setUserInfo();
+    this.updateProgress();
+  }
+
+  updateProgress() {
+    const uid = this.userInfoService.getUserInfo().uid;
```

Comment:
Will this actually work for a new user? `setUserInfo` triggers an observable, and `getUserInfo` is immediately called. I think there will be a runtime error saying `can't read uid of null` and because of that the initial status will not be set, and might have more errors later down the wokflow. Please verify.

Reply:
Right. I moved the progress update and router call within the login subscriber. 
https://github.com/armory-uci/vital/pull/20/commits/945b18dab031a186a311b77956a1753ee4b4793a